src/stdlib_experimental_io.f90

```diff
@@ -268,4 +269,39 @@ logical function whitechar(char) ! white character
 end if
 end function
 
+integer function open(filename, mode) result(u)
+! Open a file
+!
+! To open a file to read:
+!
+! u = open("somefile.txt")        # The default `mode` is "r"
+! u = open("somefile.txt", "r")
+!
+! To open a file to write:
+!
+! u = open("somefile.txt", "w")
+
+! To append to the end of the file if it exists:
+!
+! u = open("somefile.txt", "a")
+
+character(*), intent(in) :: filename
+character(*), intent(in), optional :: mode
+character(:), allocatable :: mode_
+mode_ = "r"
+if (present(mode)) mode_ = mode
+! Note: the Fortran standard says that the default values for `status` and
+! `action` are processor dependent, so we have to explicitly set them below
+if (mode_ == "r") then
+    open(newunit=u, file=filename, status="old", action="read")
```

Comment:
I would add at least `FORM=` and `ACCESS=` to explicitely show which type of files the function refers to.

```suggestion
    open(newunit=u, file=filename, status="old", action="read", &
             access='sequential', form='formatted')
```